flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/codegen/calls/BridgingFunctionGenUtil.scala

```diff
@@ -396,4 +405,183 @@ object BridgingFunctionGenUtil {
     val outputClass = outputDataType.map(_.getConversionClass).getOrElse(classOf[Unit])
     validateClassForRuntime(udf.getClass, methodName, argumentClasses, outputClass, functionName)
   }
+
+  class DefaultExpressionEvaluatorFactory(tableConfig: ReadableConfig, rexFactory: RexFactory)
+    extends ExpressionEvaluatorFactory {
+
+    override def createEvaluator(
+        function: BuiltInFunctionDefinition,
+        outputDataType: DataType,
+        args: DataType*): ExpressionEvaluator = {
+      val (argFields, call) = function match {
+        case BuiltInFunctionDefinitions.CAST | BuiltInFunctionDefinitions.TRY_CAST =>
+          Preconditions.checkArgument(args.length == 1, "Casting expects one arguments.", Seq(): _*)
+          val field = DataTypes.FIELD("arg0", args.head)
+          (
+            Seq(field),
+            unresolvedCall(function, unresolvedRef(field.getName), typeLiteral(outputDataType)))
+        case _ =>
+          val fields = args.zipWithIndex
+            .map { case (dataType, i) => DataTypes.FIELD(s"arg$i", dataType) }
+          val argRefs = fields.map(arg => unresolvedRef(arg.getName))
+          (fields, unresolvedCall(function, argRefs: _*))
+      }
+
+      createEvaluator(call, outputDataType, argFields: _*)
+    }
+
+    override def createEvaluator(
+        sqlExpression: String,
+        outputDataType: DataType,
+        args: DataTypes.Field*): ExpressionEvaluator = {
+      createEvaluator(callSql(sqlExpression), outputDataType, args: _*)
+    }
+
+    override def createEvaluator(
+        expression: Expression,
+        outputDataType: DataType,
+        args: DataTypes.Field*): ExpressionEvaluator = {
+      args.foreach(f => validateInputDataType(f.getDataType))
+      validateOutputDataType(outputDataType)
+
+      try {
+        createEvaluatorOrError(expression, outputDataType, args)
+      } catch {
+        case t: Throwable =>
+          throw new TableException(
+            s"Unable to create an expression evaluator for expression: $expression",
+            t)
+      }
+    }
+
+    private def createEvaluatorOrError(
```

Comment:
Would be great to have some sample code in comments here, similarly to what we have for the Casting functions.
Maybe refer to the example in `FunctionITCase` with the `RowEqualityScalarFunction` custom function?